src/externalsystems/OfferProvider.Library/BusinessLogic/OfferProviderBusinessLogic.cs

```diff
@@ -121,35 +121,38 @@ await _offerProviderService
             throw new ConflictException("Client should be set");
         }
 
-        if (data.ServiceAccounts.Count() > 1)
-        {
-            throw new ConflictException("There should be not more than one service account for the offer subscription");
-        }
+        List<CallbackTechnicalUserInfoData>? technicalUsersInfoData = null;
 
-        CallbackTechnicalUserInfoData? technicalUserInfoData = null;
-        if (data.ServiceAccounts.Count() == 1)
+        if (data.ServiceAccounts?.Any() == true)
         {
-            var serviceAccount = data.ServiceAccounts.First();
-            if (serviceAccount.TechnicalClientId == null)
-            {
-                throw new ConflictException($"ClientId of serviceAccount {serviceAccount.TechnicalUserId} should be set");
-            }
+            technicalUsersInfoData = [];
             async Task<string?> GetServiceAccountSecret(string technicalClientId)
             {
                 var internalClientId = await _provisioningManager.GetIdOfCentralClientAsync(technicalClientId).ConfigureAwait(ConfigureAwaitOptions.None);
                 var authData = await _provisioningManager.GetCentralClientAuthDataAsync(internalClientId).ConfigureAwait(ConfigureAwaitOptions.None);
                 return authData.Secret;
             }
-            technicalUserInfoData = new CallbackTechnicalUserInfoData(
-                serviceAccount.TechnicalUserId,
-                serviceAccount.TechnicalUserKindId == TechnicalUserKindId.INTERNAL
+
+            foreach (var serviceAccount in data.ServiceAccounts)
+            {
+                if (serviceAccount.TechnicalClientId == null)
+                {
+                    throw new ConflictException($"ClientId of serviceAccount {serviceAccount.TechnicalUserId} should be set");
+                }
+
+                var secret = serviceAccount.TechnicalUserKindId == TechnicalUserKindId.INTERNAL
```

Comment:
```suggestion
            technicalUsersInfoData = await data.ServiceAccounts.Select(async serviceAccount =>
            {
                if (serviceAccount.TechnicalClientId == null)
                {
                    throw new ConflictException($"ClientId of serviceAccount {serviceAccount.TechnicalUserId} should be set");
                }

                var secret = serviceAccount.TechnicalUserKindId == TechnicalUserKindId.INTERNAL
                    ? await GetServiceAccountSecret(serviceAccount.TechnicalClientId).ConfigureAwait(ConfigureAwaitOptions.None)
                    : null;

                return new CallbackTechnicalUserInfoData(
                    serviceAccount.TechnicalUserId,
                    secret,
                    serviceAccount.TechnicalClientId);
            }).TasksToAsyncEnumerable(cancellationToken).ToListAsync().ConfigureAwait(false);
```
(TasksToAsyncEnumerable is in namespace Org.Eclipse.TractusX.Portal.Backend.Framework.Async)

Reply:
yeah thats right , using Linq is better. In addition, I would prefer to run it in parallel execution instead of using TasksToAsyncEnumerable. Something like this , Please suggest

```s
var tasks = data.ServiceAccounts.Select(async serviceAccount =>
                        {
                            if (serviceAccount.TechnicalClientId == null)
                            {
                                throw new ConflictException($"ClientId of serviceAccount {serviceAccount.TechnicalUserId} should be set");
                            }

                            var secret = serviceAccount.TechnicalUserKindId == TechnicalUserKindId.INTERNAL
                                ? await GetServiceAccountSecret(serviceAccount.TechnicalClientId).ConfigureAwait(ConfigureAwaitOptions.None)
                                : null;

                            return new CallbackTechnicalUserInfoData(
                                serviceAccount.TechnicalUserId,
                                secret,
                                serviceAccount.TechnicalClientId);
                        });

            technicalUsersInfoData = await Task.WhenAll(tasks).ConfigureAwait(ConfigureAwaitOptions.None);
```